### DISCUSSION TITLE
How can I make the Monaco editor fully support nodejs code assist and highlighting?

### DISCUSSION THREAD

**Discussion by aszswaz** [microsoft/monaco-editor#3693] [Q&A]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

I am planning to use Monaco editor as a code editor for my electron project.

But I encountered a small problem, the project requires users to enter nodejs code in the editor.

But I found that the Monaco editor does not support CommonJS code assist.

Here is my Webpack config:

```js
const MonacoEditorPlugin = require('monaco-editor-webpack-plugin')

module.exports = {
    plugins: [
        new MonacoEditorPlugin({
            languages: ['javascript', 'css', 'html', 'typescript', 'json']
        })
    ]
}
```

Here is my code:

```js
let codeEditor = monaco.editor.create(document.getElementById("editor-section"), {
        value: "function hello() {\n    console.log('Hello world!');\n}",
        language: "javascript",
        automaticLayout: true,
        minimap: {
          enabled: false
        },
        cursorSmoothCaretAnimation: true,
        scrollBeyondLastLine: false,
        scrollBeyondLastColumn: 0
      });
```

Here is the final effect:

![暂无图片](https://i.stack.imgur.com/eU8wJ.png)

How can I configure the Monaco editor so that it can assist nodejs code correctly?

If there is no such way, how can I turn off this CommonJS module to ES module prompt? This prompt is wrong and will mislead the user.

node version: v16.13.2

webpack version: ^5.69.0

webpack-cli version: ^4.9.2

electron: ^17.0.1

monaco-editor: ^0.32.1

**Comment by Pail4** [comment]
@aszswaz  Did you find the answer? Have the same question (:

**Comment by Neill83** [comment]
Hi, did you get it working? can't find any solution
